### ISSUE TITLE
[DX][Twig] Simplify the path and namespace configuration

### ISSUE THREAD

**Issue by javiereguiluz** [symfony/symfony#33445]
**Description**  
I have two comments related to Twig paths and namespaces.

### Improve YAML config of paths

The current format is a bit ugly/confusing because it's mandatory to add the null YAML value (`'...': ~`) even when you don't use namespaces:

```yaml
twig:
    # ...
    paths:
        '%kernel.project_dir%/vendor/acme/foo-bar/templates': ~
        '%kernel.project_dir%/email/templates': ~
        '%kernel.project_dir%/themes/default': 'theme'
```

In XML/PHP the format is OK ... but for YAML maybe we could make `paths` a simple array whose values can be strings (`'path'`) or arrays (`['path', 'namespace']`):

```yaml
twig:
    # ...
    paths:
        - '%kernel.project_dir%/vendor/acme/foo-bar/templates'
        - '%kernel.project_dir%/email/templates'
        - ['%kernel.project_dir%/themes/default', 'theme']
```

### Remove default_path? Merge it with paths?

I still don't fully understand why we define a `twig.default_path` and `twig.paths` option. Could we remove `default_path`? Define all application paths in `twig.paths`? The default one is the first one because a template is used as soon as it's found, so the first path has the most priority.

-----

In summary, these would be the changes:

**BEFORE**

```yaml
twig:
    # ...
    default_path: '%kernel.project_dir%/templates'
    paths:
        '%kernel.project_dir%/vendor/acme/foo-bar/templates': ~
        '%kernel.project_dir%/email/templates': ~
        '%kernel.project_dir%/themes/default': 'theme'
```

**AFTER**

```yaml
twig:
    # ...
    paths:
        - '%kernel.project_dir%/templates'
        - '%kernel.project_dir%/vendor/acme/foo-bar/templates'
        - '%kernel.project_dir%/email/templates'
        - ['%kernel.project_dir%/themes/default', 'theme']
```

**Comment by stof** [comment]
well, `default_path` is used to override the path registered by TwigBundle (well, the one for the main namespace, as it also registers paths for bundle namespaces in addition to that). `paths` allows registering additional paths **without replacing what TwigBundle already registers**.
Removing `default_path` cannot be done while keeping the possibility to get migrate from the old `%kernel.root_dir%/Resources/templates` path to the new `%kernel.project_dir%/templates` path (registering `%kernel.project_dir%/templates` in `paths` to make the new way works would keep both folders in a TwigBundle version before the migration, and would register the new folder twice in a TwigBundle after the migration).
My own understanding is that `default_path` is meant to be deprecated in 5.1 (as 5.0+ will already use the new default_path value by default and so omitting it would be the same).

**Comment by javiereguiluz** [comment]
Let's close this proposal because it didn't receive any support. Thanks!
